src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -511,6 +511,12 @@ private function getArgumentsAsPhp(\DOMElement $node, $name, $file, $lowercase =
                     }
                     $arguments[$key] = new TaggedIteratorArgument($arg->getAttribute('tag'));
                     break;
+                case 'binary':
+                    if (!preg_match('#^([A-Za-z0-9+/]{4})*([A-Za-z0-9+/]{4}|[A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{2}==)$#', $arg->nodeValue)) {
```

Comment:
Is that necessary? `base64_decode()` would return `false` anyway which we could check for to detect errors, can't we?

Reply:
Yes, could be done this way also. I generally tend to avoid calling methods with invalid parameters.

Reply:
I would side with @xabbuh here: This class should not care about how a valid base64 string looks like. This is an implementation detail of the `base64_decode` function.